src/xkernel_core.cpp

```diff
@@ -263,9 +275,10 @@ namespace xeus
         const nl::json& content = request.content();
         std::string code = content.value("code", "");
         int cursor_pos = content.value("cursor_pos", -1);
-
-        nl::json reply = p_interpreter->complete_request(code, cursor_pos);
-        send_reply("complete_reply", nl::json::object(), std::move(reply), c);
+        xrequest_context request_context(request.header(), c, request.identities());
+        nl::json reply = p_interpreter->complete_request(std::move(request_context), code, cursor_pos);
+        send_reply(request.identities(), "complete_reply", request.header(),
+            nl::json::object(), std::move(reply), c);
```

Comment:
Why do we need to pass the `xrequest_context` to `xinterpreter::complete_request`?

Reply:
probably we do not need it, but if a complete request would like to publish smth we would need access to the parent headers

Reply:
but, lets just not pass the context, then it's just disallowed to publish stuff from all request except the execute request